src/cli.rs

```diff
@@ -36,23 +36,31 @@ fn calculate_complexity(flags: flags::Flags) {
         .for_each(|i| files_filter.only_paths.push(i));
 
     builder.filter_entry(move |e| files_filter.matches(e.path()));
+    builder.threads(num_cpus::get());
 
-    let results = Arc::new(Mutex::new(vec![]));
-    let scorer = flags.scorer;
+    let mut results = vec![];
+    let scorer = &flags.scorer;
+
+    let (sender, receiver) = unbounded();
 
     builder.build_parallel().run(|| {
-        Box::new(|result| {
+        let sender = sender.clone();
+
+        Box::new(move |result| {
             let mut scorer = build_scorer(&scorer);
             if let Some(parsed_file) = parse_dir_entry(&mut scorer, result) {
-                let mut results = results.lock().unwrap();
-                results.push(parsed_file);
+                sender.send(parsed_file).unwrap();
```

Comment:
I've been using `expect` instead of `unwrap`, so I can see a useful error message when the unthinkable happens.

I don't feel too strongly about that here since it'd be very, very shocking if `unwrap` were to fail...